externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/PlanOnlyStatement.scala

```diff
@@ -136,10 +142,25 @@ class PlanOnlyStatement(
       case ExecutionMode =>
         val executed = spark.sql(statement).queryExecution.executedPlan
         iter = new IterableFetchIterator(Seq(Row(executed.toJSON)))
+      case LineageMode =>
+        val result = parseLineage(spark, plan)
+        iter = new IterableFetchIterator(Seq(Row(result)))
       case UnknownMode => throw unknownModeError(mode)
       case _ =>
         throw KyuubiSQLException(s"The operation mode $mode" +
           " doesn't support in Spark SQL engine.")
     }
   }
+
+  private def parseLineage(spark: SparkSession, plan: LogicalPlan): String = {
+    val analyzed = spark.sessionState.analyzer.execute(plan)
+    spark.sessionState.analyzer.checkAnalysis(analyzed)
```

Comment:
`optimized` has done this check ?

Reply:
Call the`executeAndCheck` method in the `analyzed` stage when query execution, this method includes `checkAnalysis`, but here just call the `spark.sessionState.analyzer.execute(plan)`, so `optimized` don't do this check